pytorch_lightning/trainer/trainer.py

```diff
@@ -1481,41 +1486,29 @@ def check_model_configuration(self, model: LightningModule):
             # Check val_dataloader, validation_step and validation_epoch_end
             if self.is_overridden('val_dataloader', model):
                 if not self.is_overridden('validation_step', model):
-                    raise MisconfigurationException(
-                        'You have passed in a `val_dataloader()`' ' but have not defined `validation_step()`.'
+                    rank_zero_warn(
+                        'You have passed in a `val_dataloader()`' ' but have not defined `validation_step()`.',
+                        RuntimeWarning
                     )
-                else:
-                    if not self.is_overridden('validation_epoch_end', model):
-                        rank_zero_warn(
-                            'You have defined a `val_dataloader()` and have defined a `validation_step()`,'
-                            ' you may also want to define `validation_epoch_end()` for accumulating stats.',
-                            RuntimeWarning,
-                        )
             else:
                 if self.is_overridden('validation_step', model):
                     raise MisconfigurationException(
-                        'You have defined `validation_step()`,' ' but have not passed in a `val_dataloader()`.'
-                    )
-
-        # Check test_dataloader, test_step and test_epoch_end
-        if self.is_overridden('test_dataloader', model):
-            if not self.is_overridden('test_step', model):
-                raise MisconfigurationException(
-                    'You have passed in a `test_dataloader()`' ' but have not defined `test_step()`.'
-                )
-            else:
-                if not self.is_overridden('test_epoch_end', model):
-                    rank_zero_warn(
-                        'You have defined a `test_dataloader()` and have defined a `test_step()`, you may also want to'
-                        ' define `test_epoch_end()` for accumulating stats.',
+                        'You have defined `validation_step()`,' ' but have not passed in a `val_dataloader()`.',
                         RuntimeWarning,
                     )
         else:
-            if self.testing and self.is_overridden('test_step', model):
-                raise MisconfigurationException(
-                    'You have defined `test_step()` but did not'
-                    ' implement `test_dataloader` nor passed in `.test(test_dataloader)`.'
-                )
+            # Check test_dataloader, test_step and test_epoch_end
+            if self.is_overridden('test_dataloader', model):
+                if not self.is_overridden('test_step', model):
+                    rank_zero_warn(
+                        'You have passed in a `test_dataloader()`' ' but have not defined `test_step()`.',
+                    )
+            else:
+                if self.is_overridden('test_step', model):
```

Comment:
```suggestion
                if selt.testing and self.is_overridden('test_step', model):
```
I would suggest this to be called during testing only. Let's say `test_step` is implemented but I haven't implemented `test_dataloader` and doing `.fit()`.